File: 53-maximum-subarray/53-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        
        size = len(nums)
        
        prev, summe = 0,float('-inf')
        
        for curr in nums:
            
            if (prev >=0):
                prev += curr
            
            else:
                prev = curr
            
            if(prev > summe):
                summe = prev
        
        return (summe)

File: 53-maximum-subarray/53-maximum-subarray.rs
impl Solution {
    pub fn max_sub_array(nums: Vec<i32>) -> i32 {
        let mut prev_sum = 0;
        let mut curr_sum = 0;
        let mut max_sum = i32::MIN;
        
        for num in nums{
            if (prev_sum > 0){
                prev_sum +=num;
            }
            else{
                prev_sum = num;
            }
            
            if(prev_sum > max_sum){
                max_sum = prev_sum;
            }
        }
        
        max_sum
    }
}